tests/unit/providers/agent/test_meta_reference_agent.py

```diff
@@ -0,0 +1,232 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the terms described in the LICENSE file in
+# the root directory of this source tree.
+
+from datetime import datetime
+from unittest.mock import AsyncMock
+
+import pytest
+import pytest_asyncio
+
+from llama_stack.apis.agents import (
+    Agent,
+    AgentConfig,
+    AgentCreateResponse,
+    ListAgentsResponse,
+)
+from llama_stack.apis.inference import Inference
+from llama_stack.apis.safety import Safety
+from llama_stack.apis.tools import ToolGroups, ToolRuntime
+from llama_stack.apis.vector_io import VectorIO
+from llama_stack.providers.inline.agents.meta_reference.agents import MetaReferenceAgentsImpl
+from llama_stack.providers.inline.agents.meta_reference.config import MetaReferenceAgentsImplConfig
+from llama_stack.providers.inline.agents.meta_reference.persistence import AgentInfo
+
+
+@pytest.fixture
+def mock_apis():
+    return {
+        "inference_api": AsyncMock(spec=Inference),
+        "vector_io_api": AsyncMock(spec=VectorIO),
+        "safety_api": AsyncMock(spec=Safety),
+        "tool_runtime_api": AsyncMock(spec=ToolRuntime),
+        "tool_groups_api": AsyncMock(spec=ToolGroups),
+    }
+
+
+@pytest.fixture
+def config(tmp_path):
+    return MetaReferenceAgentsImplConfig(
+        persistence_store={
+            "type": "sqlite",
+            "db_path": str(tmp_path / "test.db"),
+        },
+    )
+
+
+@pytest_asyncio.fixture
+async def agents_impl(config, mock_apis):
+    impl = MetaReferenceAgentsImpl(
+        config=config,
+        inference_api=mock_apis["inference_api"],
+        vector_io_api=mock_apis["vector_io_api"],
+        safety_api=mock_apis["safety_api"],
+        tool_runtime_api=mock_apis["tool_runtime_api"],
+        tool_groups_api=mock_apis["tool_groups_api"],
+    )
+    await impl.initialize()
+    try:
+        return impl
```

Comment:
this can be `yield impl` because otherwise isn't the shutdown getting called almost instantaneously?